src/Phinx/Db/Plan/Plan.php

```diff
@@ -442,61 +405,62 @@ protected function gatherCreates($actions)
     {
-        collection($actions)
-            ->filter(function ($action) {
-                return $action instanceof AddColumn
+        foreach ($actions as $action) {
+            if (
+                !(
+                    $action instanceof AddColumn
                     || $action instanceof ChangeColumn
                     || $action instanceof RemoveColumn
-                    || $action instanceof RenameColumn;
-            })
-            // We are only concerned with table changes
-            ->reject(function ($action) {
-                return isset($this->tableCreates[$action->getTable()->getName()]);
-            })
-            ->each(function ($action) {
-                $table = $action->getTable();
-                $name = $table->getName();
+                    || $action instanceof RenameColumn
+                )
+                || isset($this->tableCreates[$action->getTable()->getName()])
+            ) {
+                 continue;
+            }
+            $table = $action->getTable();
+            $name = $table->getName();
 
-                if ($action instanceof RemoveColumn) {
-                    if (!isset($this->columnRemoves[$name])) {
-                        $this->columnRemoves[$name] = new AlterTable($table);
-                    }
-                    $this->columnRemoves[$name]->addAction($action);
-                } else {
-                    if (!isset($this->tableUpdates[$name])) {
-                        $this->tableUpdates[$name] = new AlterTable($table);
-                    }
-                    $this->tableUpdates[$name]->addAction($action);
+            if ($action instanceof RemoveColumn) {
+                if (!isset($this->columnRemoves[$name])) {
+                    $this->columnRemoves[$name] = new AlterTable($table);
                 }
-            });
+                $this->columnRemoves[$name]->addAction($action);
+            } else {
+                if (!isset($this->tableUpdates[$name])) {
+                    $this->tableUpdates[$name] = new AlterTable($table);
+                }
+                $this->tableUpdates[$name]->addAction($action);
+            }
+        }
     }
 
     /**
      * Collects all alter table drop and renames from the given intent
      *
      * @param \Phinx\Db\Action\Action[] $actions The actions to parse
-     *
      * @return void
      */
     protected function gatherTableMoves($actions)
     {
-        collection($actions)
-            ->filter(function ($action) {
-                return $action instanceof DropTable
-                    || $action instanceof RenameTable
-                    || $action instanceof ChangePrimaryKey
-                    || $action instanceof ChangeComment;
-            })
-            ->each(function ($action) {
-                $table = $action->getTable();
-                $name = $table->getName();
+        foreach ($actions as $action) {
+            if (
+                !(
```

Comment:
This does not read well.

Reply:
I'm not clear what you would suggest here. I would agree initially with just modifying it so that the `!(` are on the opening line:
```php
            if (!(
                $action instanceof DropTable
                || $action instanceof RenameTable
                || $action instanceof ChangePrimaryKey
                || $action instanceof ChangeComment
            )) {
                continue;
            }
```

An alternative would be to move the negation down a level:

```php
            if (
                !($action instanceof DropTable)
                && !($action instanceof RenameTable)
                && !($action instanceof ChangePrimaryKey)
                && !($action instanceof ChangeComment)
            ) {
                continue;
            }
```

The only other idea I've got would be to utilize an outer array_filter on `$actions`, except I'd want to avoid that in-favor of the inline option to save on array cycles.

Reply:
I would prefer the 2nd implementation. it seems like more text, but it's easier to read.

Reply:
alright, done.